src/connection-manager/index.ts

```diff
@@ -688,34 +702,42 @@ export class DefaultConnectionManager extends EventEmitter<ConnectionManagerEven
     )
   }
 
-  async acceptIncomingConnection (maConn: MultiaddrConnection) {
-    try {
-      // check deny list
-      const denyConnection = this.deny.some(ma => {
-        return maConn.remoteAddr.toString().startsWith(ma.toString())
-      })
+  async acceptIncomingConnection (maConn: MultiaddrConnection): Promise<boolean> {
+    // check deny list
+    const denyConnection = this.deny.some(ma => {
+      return maConn.remoteAddr.toString().startsWith(ma.toString())
+    })
 
-      if (denyConnection) {
-        throw errCode(new Error('connection remote address was in deny list'), codes.ERR_CONNECTION_DENIED)
-      }
+    if (denyConnection) {
+      log('connection remote address was in deny list')
+      return false
+    }
 
-      // check allow list
-      const allowConnection = this.allow.some(ma => {
-        return maConn.remoteAddr.toString().startsWith(ma.toString())
-      })
+    // check allow list
+    const allowConnection = this.allow.some(ma => {
+      return maConn.remoteAddr.toString().startsWith(ma.toString())
+    })
 
-      if (allowConnection) {
-        return
-      }
+    if (allowConnection) {
+      return true
+    }
 
-      if (this.getConnections().length < this.opts.maxConnections) {
-        return
+    if (maConn.remoteAddr.isThinWaistAddress()) {
+      const host = maConn.remoteAddr.nodeAddress().address
+
+      try {
+        await this.inboundConnectionRateLimiter.consume(host, 1)
+      } catch {
+        log('inboundConnectionThreshold exceeded by host %s', host)
+        return false
       }
+    }
 
-      throw errCode(new Error('maxConnections exceeded'), codes.ERR_TOO_MANY_CONNECTIONS)
-    } catch (err) {
-      await maConn.close()
-      throw err
+    if (this.getConnections().length < this.opts.maxConnections) {
+      return true
     }
+
+    log('maxConnections exceeded')
+    return false
```

Comment:
The caller no longer knows why this connection wasn't accepted. Probably okay, but maybe better to return an error?

Reply:
It logs why the connection was accepted - it's for incoming connections only so it's really just informational for anyone watching the logs.

More idiomatic to throw than return an error but after the conversation [here](https://github.com/libp2p/js-libp2p-interfaces/pull/295#discussion_r986720181) I changed it to return a boolean instead.